x-pack/plugins/observability_solution/logs_explorer/common/data_views/models/data_view_descriptor.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DataViewListItem } from '@kbn/data-views-plugin/common';
+import { TIMESTAMP_FIELD } from '../../constants';
+import { DataViewSpecWithId } from '../../dataset_selection';
+import { DataViewDescriptorType } from '../types';
+import { buildIndexPatternRegExp } from '../utils';
+
+type Whitelist = Array<string | RegExp>;
+
+const LOGS_WHITELIST: Whitelist = [
+  buildIndexPatternRegExp(['logs', 'auditbeat', 'filebeat', 'winbeat']),
+  // Add more strings or regex patterns as needed
+];
+
+export class DataViewDescriptor {
+  id: DataViewDescriptorType['id'];
+  dataType: DataViewDescriptorType['dataType'];
+  name: DataViewDescriptorType['name'];
+  namespaces: DataViewDescriptorType['namespaces'];
+  title: DataViewDescriptorType['title'];
+  type: DataViewDescriptorType['type'];
+
+  private constructor(dataViewDescriptor: DataViewDescriptorType) {
+    this.id = dataViewDescriptor.id;
+    this.dataType = dataViewDescriptor.dataType;
+    this.name = dataViewDescriptor.name;
+    this.namespaces = dataViewDescriptor.namespaces;
+    this.title = dataViewDescriptor.title;
+    this.type = dataViewDescriptor.type;
+  }
+
+  getFullTitle() {
+    return this.name;
+  }
+
+  toDataviewSpec(): DataViewSpecWithId {
+    return {
+      id: this.id,
+      name: this.name,
+      timeFieldName: TIMESTAMP_FIELD,
+      title: this.title,
+    };
+  }
+
+  toPlain() {
+    return {
+      id: this.id,
+      dataType: this.dataType,
+      name: this.name,
+      title: this.title,
+    };
+  }
+
+  public static create(dataViewListItem: DataViewListItem) {
+    const name = dataViewListItem.name ?? dataViewListItem.title;
+    const dataType = DataViewDescriptor.#extractDataType(dataViewListItem.title);
```

Comment:
**note**: when creating an instance, we want to make sure:
- the name associated with the data view is not undefined and fallbacks instead to the title
- the dataType is extracted once when constructing the instance to prevent repetitive calls of the `extractDataType` expensive method.